### PR TITLE
Convert private variables to function scope.

### PR DESCRIPTION
**Pull request by s-perron**
When a private variable is used in a single function, it can be
converted to a function scope variable in that function.  This adds a
pass that does that.  The pass can be enabled using the option
`--private-to-local`.

This transformation allows other transformations to act on these
variables.

Also moved `FindPointerToType` from the inline class to the type manager.

Fixes issue #1086.

### CODE REVIEW

File: source/opt/private_to_local_pass.cpp [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+const int kSpvTypePointerTypeIdInIdx = 1;
+}  // namespace
+
+namespace spvtools {
+namespace opt {
+
+Pass::Status PrivateToLocalPass::Process(ir::IRContext* c) {
+  InitializeProcessing(c);
+  bool modified = false;
+
```
Comment (alan-baker) [inline source/opt/private_to_local_pass.cpp]: The semantics on the lifetime of a private scope variable outside of shaders models are unclear. You should predicate this pass on the shader capabality.

Comment (s-perron) [reply source/opt/private_to_local_pass.cpp]: From the spec, I think that private variable only exist in shader modules.  I'll add the check, so we don't do pointless work.

File: source/opt/private_to_local_pass.cpp [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "private_to_local_pass.h"
+
+#include "ir_context.h"
+
+namespace {
+const uint32_t kVariableStorageClassInIdx = 0;
+const int kSpvTypePointerTypeIdInIdx = 1;
```
Comment (dnovillo) [inline source/opt/private_to_local_pass.cpp]: uint32_t?

File: source/opt/private_to_local_pass.cpp [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+  }
+
+  modified = !variables_to_move.empty();
+  for (auto p : variables_to_move) {
+    MoveVariable(p.first, p.second);
+  }
+
+  return (modified ? Status::SuccessWithChange : Status::SuccessWithoutChange);
+}
+ir::Function* PrivateToLocalPass::FindLocalFunction(
```
Comment (dnovillo) [inline source/opt/private_to_local_pass.cpp]: Blank line above for readability.

File: source/opt/private_to_local_pass.cpp (hunk 2 above)
Comment (s-perron) [reply source/opt/private_to_local_pass.cpp]: done

File: source/opt/private_to_local_pass.cpp (hunk 3 above)
Comment (s-perron) [reply source/opt/private_to_local_pass.cpp]: done

**Pull request merged**